hack/update-codecgen.sh

```diff
@@ -52,7 +98,9 @@ for generated_file in ${generated_files}; do
 	rm -f "${generated_file}"
 done
 
-for generated_file in ${generated_files}; do
+# Generate files in the dependency order.
+for current in ${index[@]}; do
```

Comment:
Rather than sorting everything ahead of time, I think it might be simpler to just do something along the lines of:
```
while not done {
  for each ungenerated file {
    if file has no dependencies, generate it
  }
  if no files were generated, error
}
```